R/pwr_1p.R

```diff
@@ -1,702 +1,1026 @@
+    gettextf("The design's power for detecting effects of %1$s%2$s is thus %3$s.", alt_text, d, round(power, 3)),
+    "</p>"
+  )
+
+
+  html[["text"]] <- str
+}
+
+
+#### Populate table ----
+.populatePowerTabTest1P <- function(jaspResults, options, r, lst) {
+  table <- jaspResults[["powertab"]]
+
+  calc <- options$calculation
+  n <- ifelse(calc == "sampleSize", r$n, lst$n)
+  p0 <- lst$p0
+  if (calc == "effectSize") p1 <- ifelse(lst$alt == "two.sided" && options$effectDirection == "less", r$p[2], r$p[1]) else p1 <- lst$p1
+  d <- abs(2 * (asin(sqrt(p1)) - asin(sqrt(p0))))
+  power <- ifelse(calc == "power", r$power, lst$pow)
+  alpha <- ifelse(calc == "alpha", r$alpha, lst$alpha)
+  alt <- lst$alt
+
+  if (calc == "effectSize") {
+    table$addColumns(list(comparisonProportion = p1))
+    table$addColumns(list(effectSize = d))
+  } else {
+    row <- list()
+    row[[calc]] <- r[[switch(calc,
+      "sampleSize" = "n",
+      calc
+    )]]
+    table$addColumns(row)
+  }
+
+  if (calc == "sampleSize") {
+    if (round(.pwrPTest(n = n, p0 = p0, p = p1, sig.level = alpha, alternative = alt)$power, 3) == 1) {
+      table$addFootnote(gettext("Due to the rounding of the sample size, the actual power can deviate from the target power. <b>Actual power: >0.999"))
+    } else {
+      table$addFootnote(gettextf(
+        "Due to the rounding of the sample size, the actual power can deviate from the target power. <b>Actual power: %1$s</b>",
+        round(.pwrPTest(n = n, p0 = p0, p = p1, sig.level = alpha, alternative = alt)$power, 3)
+      ))
+    }
+  }
+}
+
+.populatePowerESTabTest1PTest1P <- function(jaspResults, options, r, lst) {
+  html <- jaspResults[["tabText"]]
+  if (is.null(html)) {
+    html <- createJaspHtml()
+    html$dependOn(c("test", "text"))
+    html$position <- 3
+    jaspResults[["tabText"]] <- html
+  }
+
+  ## Get options from interface
+  calc <- options$calculation
+  n <- ifelse(calc == "sampleSize", r$n, lst$n)
+  p0 <- lst$p0
+  if (calc == "effectSize") p1 <- ifelse(lst$alt == "two.sided" && options$effectDirection == "less", r$p[2], r$p[1]) else p1 <- lst$p1
+  d <- abs(2 * (asin(sqrt(p1)) - asin(sqrt(p0))))
+  power <- ifelse(calc == "power", r$power, lst$pow)
+  alpha <- ifelse(calc == "alpha", r$alpha, lst$alpha)
+  alt <- lst$alt
+
+  n_text <- gettextf("a sample size of %s", n)
+
+  tail_text <- ifelse(alt == "two.sided",
+    gettext("two-sided"),
+    gettext("one-sided")
   )
-)
+
+  if (calc == "sampleSize") {
+    str <- gettextf(
+      "We would need %1$s to reliably (with probability greater than or equal to %2$s) detect an effect size of <i>|h|%3$s</i>%4$s, assuming a %5$s criterion for detection that allows for a maximum Type I error rate of <i>\u03B1=</i>%6$s.",
```

Comment:
html tahs and unicode symbols should be passed into gettextf as arguments; I assume we don't want the translators with changing e.g., `<i>` to `<br>`!